### PR TITLE
Feature/us6 list of interactions

### LINKED ISSUE #6: Show a list of all (social) interactions
## Value proposition

As a user
I want to see a list of all (social) interactions
So that I know which interactions I had in the past with friends and family

## Description

Image

## Acceptance criteria
- [x] Every (social) interaction is in a list item below each other
- [x] The list has a heading that says "Interactions"
- Each list item displays the following information:
  - [x] Icon that shows the type of the interaction. Types can be:
    - call
    - incoming-call
    - outgoing-call
    - video-conference
    - party
    - other
  - [x] Contacts, that are referenced as participants of the interaction (alphabetically by first name)
    - [x] Participants last names are shortened (no unnecessary spaces) in order to have room for multiple contacts
  - [x] Date, the interaction took place
- [x] Items are sorted by interactionDate on default, latest on top

### PR DESCRIPTION
**Pull request by josch87**
- closes #6 
- have a look at the page /interactions
- Navigating between pages (/contacts & /interactions) is part of #7

### CODE REVIEW

File: components/InteractionListItem/InteractionListItem.js [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+          alt={`Icon of ${interaction.type.toLowerCase()} interaction`}
+        />
+        <DetailsContainer>
+          <ParticipantsContainer>
+            {sortedParticipants.map((participant, index) => {
+              if (index < sortedParticipants.length - 1) {
+                return getShortName(participant) + ", ";
+              }
+              return getShortName(participant);
+            })}
```
Comment (djfarly) [inline components/InteractionListItem/InteractionListItem.js:46]: you can make this simpler with `join(', ')`
```suggestion
            {sortedParticipants
              .map((participant) => getShortName(participant))
              .join(', ')}
```

**Review by djfarly — APPROVED** [review]
Hej Aljoscha.

Looking good! I was taking some time to actually find the page… but sometimes reading can help. :)

The code looks good, just some stylistic thing.

Cheers, Jan

File: components/InteractionList/InteractionList.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  return (
+    <ul>
+      {interactions.map((interaction) => {
+        return (
+          <Fragment key={interaction.id}>
+            <InteractionListItem
+              interaction={interaction}
+              contacts={contacts}
+            />
+          </Fragment>
```
[the remark this answers is not in the document]
Comment (josch87) [reply components/InteractionList/InteractionList.js:14]: Got rid of fragment in InteractionListItem and ContactListItem components.

File: components/InteractionListItem/InteractionListItem.js (hunk 1 above)
Comment (josch87) [reply components/InteractionListItem/InteractionListItem.js:46]: Nice. CONCAT_WS in JavaScript :)
Works.

**Pull request merged**